### ISSUE TITLE
Insert a Horizontal Rule in a Document

### ISSUE THREAD

**Issue by nickian** [ONLYOFFICE/DocumentServer#679]
I'm surprised I haven't been able to find anything on this, but is there a way to insert a horizontal rule in the Document editor? If this feature does not exist, can it be added to a future version? Having to manually draw a line shape is tedious. Thanks.

**Comment by KirillovIlya** [comment]
Hello. What do you mean by “horizontal rule”? Whether this is the upper or lower border of the paragraph, if so, you can add it through the advanced settings for the paragraph.

**Comment by nickian** [comment]
Yes, I just mean the equivalent of an HTML `<hr>`. The bottom border does work, or I can manually draw a line shape, but those are a little tedious if you have to use this a lot. It would be nice it it were just a preset of a separate line shape that was inserted as a block element with 100% width for the page. See how Microsoft Word does this:

<img width="977" alt="Screen Shot 2019-11-01 at 10 53 10 AM" src="https://user-images.githubusercontent.com/590678/68041328-f313d280-fc95-11e9-89a0-bbe4f9a7fb08.png">

**Comment by Rita-Bubnova** [comment]
Thank you, @nickian.
I can confirm - this is known issue 18816 in our private tracker.

**Comment by oMrSmith** [comment]
+1

**Comment by AJeschor** [comment]
+1
I would like this feature as well

**Comment by Macr0Nerd** [comment]
+1
Would be quite nice to have

**Comment by dmi3s** [comment]
Three years have passed. But the horizontal line is still not possible to insert into the document.

**Comment by oMrSmith** [comment]
It is possible, just not in a convenient way. This problem however is a good reason for me
and I'd suppose for many others too, not to use this great software at all.

**Comment by nickian** [comment]
I would think this would be relatively trivial to implement. Just about everything has this feature, even markdown with `***`

***

See.

**Comment by pflegende** [comment]
... realy!
I'v searched the menues and than the docs half a hour.
😵‍💫 

So I walk back to libre ...
:(

**Comment by Rita-Bubnova** [comment]
Sorry, no update.

**Comment by orion-v99** [comment]
Possible Update Here
1. Highlight desired section, right click and look for Paragraph Advance Settings
![image](https://github.com/ONLYOFFICE/DocumentServer/assets/111268860/51727c56-4ede-42b0-bd39-6716b14483c9)

2. In the Borders & Fill area: deselect any unwanted borders only showing the bottom border present in the diagram as so (this DOES NOT create a table. Simply a line underneath)
![image](https://github.com/ONLYOFFICE/DocumentServer/assets/111268860/ce7cb278-4cf4-44fd-967f-0568fd63322f)

**Comment by sesu-tech** [comment]
> Possible Update Here
> 
>     1. Highlight desired section, right click and look for Paragraph Advance Settings
>        ![image](https://user-images.githubusercontent.com/111268860/281605538-51727c56-4ede-42b0-bd39-6716b14483c9.png)
> 
>     2. In the Borders & Fill area: deselect any unwanted borders only showing the bottom border present in the diagram as so (this DOES NOT create a table. Simply a line underneath)
>        ![image](https://user-images.githubusercontent.com/111268860/281605925-ce7cb278-4cf4-44fd-967f-0568fd63322f.png)
> 
>     3. Horizontal Rule, Line Break, Horizontal Line
>        ![image](https://user-images.githubusercontent.com/111268860/281606274-3ff1ca1a-a672-46e5-a9a7-acf23fac8997.png)

This works for me! A dedicated horizontal line button would be sweet, but just adding a border in empty space seems to do the job. :)

**Comment by nickian** [comment]
That's not a horizontal rule. You could also simply draw a straight line from the shapes menu, but that's not what I had in mind. Like every other word processor, you should be able to go to the insert menu and select "horizontal rule," where it automatically inserts a 100% width 1px block.

**Comment by ImagoDeo** [comment]
+1 for a simple horizontal rule.

**Comment by genderneutralnoun** [comment]
I tried copy and pasting a html `<hr>` into my document and it didn't work. I'd appreciate it if anyone who knew a workaround even if it's tedious could share it with me; it's very essential to me. I've been using line breaks since I post my stories online and line breaks are easily read by screen readers as a brief silence (at least, from what I've heard) and thus a more accessible solution for creating breaks in a story. This is literally an accessibility feature; the fact that it's gone this long unimplemented is very egregious IMO.

**Comment by gitthangbaby** [comment]
I am looking for a solution for years, i start every document with address and line under it. Tired of painting it with permanent pen.

**Comment by nucspl** [comment]
it's 2025 and we still don't have this .-.

**Comment by oMrSmith** [comment]
It can't be that hard to implement such a basic and simple feature, can it?

**Comment by nickian** [comment]
Yeah, I can't believe this still isn't a feature. This should be considered a core formatting option. Literally, every other document editor I've used can do this. Even markdown can do this with `---`

---

^ See.

Speaking of which, that should also be a shortcut in the Document Editor (three dashes should automatically draw a horizontal ilne).

**Comment by gitthangbaby** [comment]
---

type "-" "-" "-" "Enter" boom, whole line appears once implemented

I bought M$ Office to get this function back as almost every document starts with this. Will patiently wait tho! Thanks.

**Comment by eiinu** [comment]
It is a good idea that the editor converts `---` into a horizontal line whether the line is a paragraph border or a line shape. I can't wait for this feature.

**Comment by Kirshi912** [comment]
Any update about this?

**Comment by Rita-Bubnova** [comment]
Sorry, currently we have no ETA.
I'm just a QA specialist, I'm not in position to gurantee any dates. You better contact sales@onlyoffice.com.

**Comment by ItzTas** [comment]
any updates on this?

**Comment by BOplaid** [comment]
How the heck is this not implemented yet

**Comment by nickian** [comment]
Wow, I created this in 2019 and still nothing? We have time implement "infinite AI capabilities," but not a basic horizontal rule that almost everyone uses?

**Comment by vernalan** [comment]
Its 2026, and still nothing?

**Comment by LinuxinaBit** [comment]
I just found out about this and I can't believe it...
I genuinely don't understand how OnlyOffice can be missing such _basic_ stuff!

**Comment by TatianaKirillova** [comment]
Hello @nickian I'm happy to report that the issue is fixed and will be included in the next release. Don't miss updates. Have a nice day!

**Comment by BOplaid** [comment]
> Hello [@nickian](https://github.com/nickian) I'm happy to report that the issue is fixed and will be included in the next release. Don't miss updates. Have a nice day!

At last, after
# **MORE THAN _6 YEARS_**

**Comment by Kirshi912** [comment]
> Hello [@nickian](https://github.com/nickian) I'm happy to report that the issue is fixed and will be included in the next release. Don't miss updates. Have a nice day!

They added it now, but i don't know how to insert it. What the keyboard shortcut for it?

**Comment by vernalan** [comment]
> They added it now, but i don't know how to insert it. What the keyboard shortcut for it?

I can't seem to find how to insert it either 😞

**Comment by Rita-Bubnova** [comment]
DocumentServer v9.4.0 is released so I close this issue. Feel free to comment or reopen it if you got further questions.

**Comment by Kirshi912** [comment]
> DocumentServer v9.4.0 is released so I close this issue. Feel free to comment or reopen it if you got further questions.

Please tell me how to insert that horizontal line?

**Comment by TatianaKirillova** [comment]
Hello @Kirshi912
To add a horizontal line, please go to the Home -> Borders -> Horizontal line menu

<img width="497" height="505" alt="Image" src="https://github.com/user-attachments/assets/98944b1a-ac2d-40f8-81c4-3e460a62beee" />
